src/java/TopoChromExperiment.java

```diff
@@ -1,7 +1,126 @@
 package java;
 
+import org.bitcoinj.core.*;
+import org.bitcoinj.core.listeners.PreMessageReceivedEventListener;
+
+import java.net.Inet4Address;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
 /**
  * Created by snakecharmer1024 on 7/19/16.
  */
 public class TopoChromExperiment {
+
+    private static PermutationManager manager;
+
+    private static PeerGroup listeningPeers;
+
+    private static List<PeerGroup> sendingPeers;
+
+    private static NetworkParameters parameters;
+
+    private static List<Transaction> txs;
+
+    private static int waitTimeMilli = 100;
+
+    private static void runExperiment(Transaction tx)
+    {
+        // randomly select a sending peer to send to
+        // this peer is the data label for that transaction "tx"
+        Random random = new Random();
+        int peerIndex = random.nextInt();
+        PeerGroup entryPeer = sendingPeers.get(peerIndex % sendingPeers.size());
+        manager.addLabelledTx(tx.getHash(), peerIndex);
+        entryPeer.broadcastTransaction(tx);
+    }
+
+    private static ArrayList<Transaction> createTxList()
+    {
+        // TODO TerraFlux here?
```

Comment:
@Terraflux Not sure if these should be made all at once or one by one... if we do it one by one, we might run into issues with transactions having unconfirmed inputs when we send them around. I don't know if that will affect the routing of them or not...we should probably make sure outputs have at least one confirmation before including them in a transaction experiment.